### PR TITLE
[Cartographie] [Détection Machines] [Approche Finale] Clean

### PR DESCRIPTION
**Pull request by tdanel**
This change is [Reviewable](https://reviewable.io/reviews/pyroteam/robocup-pkg/125)

### CODE REVIEW

**Comment by vcoelen** [comment]
Reviewed 35 of 40 files at r1.
Review status: 35 of 40 files reviewed at latest revision, 1 unresolved discussion.

---

_[messages/deplacement_msg/msg/OccupancyGridFloat.msg, line 1 [r1]](https://reviewable.io:443/reviews/pyroteam/robocup-pkg/125#-KN2xbjBUtP71HIS0gB4:-KN2xbjC5sICfAjXIUGS:641463553) ([raw file](https://github.com/pyroteam/robocup-pkg/blob/c5bc08857c74de0c4e2b6cab87f4778bd52b9d73/messages/deplacement_msg/msg/OccupancyGridFloat.msg#L1)):_

> ```
> std_msgs/Header header
> ```

Ne pas supprimer pliz :)

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pyroteam/robocup-pkg/125)_

**Comment by tdanel** [comment]
Reviewed 2 of 40 files at r1.
Review status: 36 of 40 files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pyroteam/robocup-pkg/125)_

**Comment by ValentinVERGEZ** [comment]
Reviewed 3 of 40 files at r1, 1 of 3 files at r2.
Review status: 38 of 40 files reviewed at latest revision, 2 unresolved discussions.

---

_[navigation/final_approach/src/Segment.cpp, line 37 [r2]](https://reviewable.io:443/reviews/pyroteam/robocup-pkg/125#-KN9L4vhDzcY9P7pxxqa:-KN9L4vi1puYm2j-HJ_Z:1201526854) ([raw file](https://github.com/pyroteam/robocup-pkg/blob/a003b7d54cd508c5ba85b37d1890c29eb56cb6fc/navigation/final_approach/src/Segment.cpp#L37)):_

> ``` cpp
> Segment::~Segment(){}
> 
> geometry_msgs::Pose2D Segment::linearRegression(std::vector<Point> pointsVector)
> ```

Désolé, je ne peux pas valider ça sans plus y regarder. Ça me semble correct, oui, mais ça suffit pas.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pyroteam/robocup-pkg/125)_

File: navigation/final_approach/src/Segment.cpp [hunk 1]
```diff
@@ -115,20 +69,16 @@ geometry_msgs::Pose2D Segment::linearRegression(std::vector<Point> pointsVector)
...
-	m_correlation = correl * correl;
+	ROS_DEBUG_NAMED("segmentConstruction",
+                  "Coeff correlation: %4.2f | Droite (%5.3f, %5.3f) Angle: %5.3f | Pente: %4.2f",
+                  correl, m_middlePoint.x, m_middlePoint.y, m_angle, slope);
 
-	m_gradient = atan2(varianceY, covarianceXY);
-	m_gradient = fmod(m_gradient, 2*M_PI);
-	// m_angle = m_gradient; // TODO: Remove
-	m_gradient = m_gradient - M_PI_2;
+	m_correlation = correl * correl;
```
Comment (ValentinVERGEZ) [inline navigation/final_approach/src/Segment.cpp]: Tu ne retournerais pas déjà correl \* correl dans ta fonction common ?

**Comment by tdanel** [comment]
Reviewed 1 of 40 files at r1, 2 of 3 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

_[navigation/final_approach/src/Segment.cpp, line 76 [r2]](https://reviewable.io:443/reviews/pyroteam/robocup-pkg/125#-KMy-o4EhnIlRnYWIAAI-r2-76:-KNGSMPGRHACyhvHCWh3:1411520145) ([raw file](https://github.com/pyroteam/robocup-pkg/blob/a003b7d54cd508c5ba85b37d1890c29eb56cb6fc/navigation/final_approach/src/Segment.cpp#L76)):_
<details><summary>

<i>Previously, ValentinVERGEZ (Valentin VERGEZ) wrote…</i></summary>



> Tu ne retournerais pas déjà correl \* correl dans ta fonction common ?
> </details>
> Oui, petite faute d'adaptation :)

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pyroteam/robocup-pkg/125)_

**Comment by tdanel** [comment]
A merger

**Comment by tdanel** [comment]
To be reviewed ! :D

**Comment by tdanel** [comment]
Up

**Comment by tdanel** [comment]
Val is ok

**Pull request merged**